impl/src/main/java/org/ehcache/internal/store/heap/OnHeapStore.java

```diff
@@ -1227,27 +1235,31 @@ private void expireMapping(final K key, final ValueHolder<V> value) {
 
   private OnHeapValueHolder<V> cloneValueHolder(K key, ValueHolder<V> valueHolder, long now, Duration expiration) {
     V realValue = valueHolder.value();
+    boolean veto = checkVeto(CacheStoreHelper.cacheEntry(key, valueHolder, timeSource));
     if(valueCopier instanceof SerializingCopier) {
-      return new SerializedOnHeapValueHolder<V>(valueHolder, realValue, ((SerializingCopier)valueCopier).getSerializer(), now, expiration);
+      return new SerializedOnHeapValueHolder<V>(valueHolder, realValue, veto, ((SerializingCopier)valueCopier).getSerializer(), now, expiration);
     } else {
-      return new CopiedOnHeapValueHolder<V>(valueHolder, realValue, valueCopier, now, expiration);
+      return new CopiedOnHeapValueHolder<V>(valueHolder, realValue, veto, valueCopier, now, expiration);
     }
   }
 
-  private OnHeapValueHolder<V> makeValue(V value, long creationTime, long expirationTime, Copier<V> valueCopier) {
-    if(valueCopier instanceof SerializingCopier) {
-      return makeSerializedValue(value, creationTime, expirationTime, ((SerializingCopier)valueCopier).getSerializer());
+  private OnHeapValueHolder<V> makeValue(K key, V value, long creationTime, long expirationTime, Copier<V> valueCopier) {
+    boolean veto = checkVeto(CacheStoreHelper.cacheEntry(key, value, creationTime, OnHeapValueHolder.TIME_UNIT));
```

Comment:
With the currently discussed changes, there are only two reasons the veto predicate would be called:
 - during initial insertion
 - during faulting

In the 1st case, this is useless as the hit rate will always be invalid since the mapping is currently being inserted. Regarding the 2nd case, we could add the extra complexity but I'd rather follow the route of my note (making the veto flag common across all tiers) which would stop calling the veto predicate during faulting.